arrow-effects/src/test/kotlin/arrow/effects/data/DurationTest.kt

```diff
@@ -0,0 +1,36 @@
+package arrow.effects.data
+
+import arrow.effects.Duration
+import arrow.test.UnitSpec
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import org.junit.runner.RunWith
+import java.util.concurrent.TimeUnit
+
+/**
+ *
+ */
+@RunWith(KTestJUnitRunner::class)
+class DurationTest : UnitSpec() {
+
+    init {
+        "plus should be commutative" {
+            forAll(Gen.long(), TimeUnitGen(), Gen.long(), TimeUnitGen()) { i: Long, u: TimeUnit, j: Long, v: TimeUnit ->
+                val a = Duration(i, u)
+                val b = Duration(j, v)
+                a + b == b + a
+            }
+        }
+    }
+
+}
+
+private class TimeUnitGen : Gen<TimeUnit> {
```

Comment:
Can you please move this to: https://github.com/arrow-kt/arrow/blob/master/arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt